app/Admin/src/Http/Controllers/Auth/AuthController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Admin\Http\Controllers\Auth;
+
+use App\Admin\Http\Requests\Auth\LoginRequest;
+use App\Admin\Http\Requests\Auth\LogoutRequest;
+use App\Admin\Http\Resources\Auth\AdminResource;
+use App\Models\User;
+use App\Platform\Enums\RoleName;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Validation\ValidationException;
+use Knuckles\Scribe\Attributes\Authenticated;
+use Knuckles\Scribe\Attributes\BodyParam;
+use Knuckles\Scribe\Attributes\Endpoint;
+
+class AuthController
+{
+    /**
+     * @throws ValidationException
+     */
+    #[Endpoint('Admin/Login', <<<'DESC'
+    This endpoint enable users with admin role to sign in
+ DESC)]
+    #[BodyParam('email', 'password', required: true)]
+    public function login(LoginRequest $request): AdminResource
+    {
+        $credentials = $request->only('email', 'password');
+
+        $user = User::where('email', $request->email)->first();
+
+        if ($user && $user->hasRole(RoleName::ADMIN)) {
```

Comment:
You can reverse this if check and make it defensive so that you return the validation exception first if the person trying to sign in does not have the admin role and move the happy path code below to avoid all of these if nested statements and code within them.

Reply:
Also you can append the authentication attempt here.

It may look something like:

See if the user was found and if it does not have the role or if the authentication attempt is not true.